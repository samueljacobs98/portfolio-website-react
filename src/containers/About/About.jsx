import "./About.scss";
import profileImg from "../../assets/images/profile-img.jpg";
import aboutData from "../../assets/data/aboutData";
import sun from "../../assets/svgs/sun-side.svg";

const About = () => {
  return (
    <div className="about" id="about">
      <div className="about__content-container">
        <img className="about__img" src={profileImg} alt="" />
        <div className="about__container">
          <h2 className="about__item-title">About Me</h2>
          <div className="about__container-item">
            <p className="about__item-text">{aboutData.summary1}</p>
            <br />
            <p className="about__item-text">{aboutData.summary2}</p>
          </div>
        </div>
      </div>
      <img className="about__sun" src={sun} alt="the sun" />
    </div>
  );
};

export default About;
