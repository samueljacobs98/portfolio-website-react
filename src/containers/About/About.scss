@use "../../assets/scss/variables" as *;

.about {
  position: relative;
  width: 100%;
  min-height: 100vh;
  padding-bottom: 0;
  display: flex;
  align-items: center;

  &__content-container {
    width: 100%;
    display: flex;
    justify-content: space-around;
    .about__img {
      max-width: 300px;
      align-self: center;
    }
    .about__container {
      align-self: center;
      display: grid;
      grid-template-columns: repeat(auto-fit, 300px);
      gap: $gap-project;
      width: 620px;
      &-item {
        max-width: 300px;
      }
    }
  }
  &__next {
    position: absolute;
    bottom: 0;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;
    &-btn:hover {
      cursor: pointer;
    }
  }
  &__sun {
    position: absolute;
    left: 0;
    top: -50%;
    z-index: -1;
  }
}

@media screen and (max-width: 960px) {
  .about {
    &__content-container {
      .about__img {
        display: none;
      }
    }
  }
}

@media screen and (max-width: 660px) {
  .about {
    margin-top: $margin-about;
    .about__content-container {
      .about__container {
        justify-content: center;
      }
    }
  }
}
