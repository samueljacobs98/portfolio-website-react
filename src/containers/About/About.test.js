import { render } from "@testing-library/react";
import About from "./About";

test("should match snapshot", () => {
  const { container } = render(<About />);
  expect(container).toMatchSnapshot();
});
