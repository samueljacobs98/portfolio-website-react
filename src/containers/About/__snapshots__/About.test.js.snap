// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="about"
    id="about"
  >
    <h2
      class="about__item-title"
    >
      About Me
    </h2>
    <div
      class="about__container"
    >
      <img
        alt=""
        class="about__img"
        src="profile-img.jpg"
      />
      <div
        class="about__summary-container"
      >
        <p
          class="about__item-text"
        >
          I am a first-class mechanical engineering graduate and junior full-stack engineer. I chose to study mechanical engineering because I love solving problems, and have gained both technical skills and transferable skills. My degree developed my problem-solving abilities for problems at any scale, in a team and independently.
        </p>
        <br />
        <p
          class="about__item-text"
        >
          I chose to pursue a career in tech as it is an exciting and rewarding combination of problem solving and design. My focus is delivering solutions from effective critical and lateral thinking, using data driven insights to carve human-centred solutions.
        </p>
      </div>
    </div>
    <img
      alt="the sun"
      class="about__sun"
      src="sun-side.svg"
    />
  </div>
</div>
`;
